### ISSUE TITLE
An error occurs when using `Prism::Translation::Parser`

### ISSUE THREAD

**Issue by koic** [ruby/prism#2302]
`invalid byte sequence in UTF-8` error occurs when parsing the following code with `Prism::Translation::Parser.parse`:

```ruby
# It can inspect non-UTF-8 encoded source with proper encoding comment.
<<~RUBY.force_encoding('iso-8859-15')
  # coding: ISO-8859-15
  # Euro symbol: \xa4
RUBY
```

## Expected

No errors.

## Actual

The following `ArgumentError` occurs:

```console
$ cat heredoc_with_force_encoding.rb
# It can inspect non-UTF-8 encoded source with proper encoding comment.
<<~RUBY.force_encoding('iso-8859-15')
  # coding: ISO-8859-15
  # Euro symbol: \xa4
RUBY

$ ruby -rprism -rprism/translation/parser -e 'p Prism::Translation::Parser.parse_file("heredoc_with_force_encoding.rb")'
/Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/prism-0.19.0/lib/prism/translation/parser/compiler.rb:1748:in `count': invalid byte sequence in UTF-8 (ArgumentError)

              if part.is_a?(StringNode) && part.unescaped.count("\n") > 1
                                                                ^^^^
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/prism-0.19.0/lib/prism/translation/parser/compiler.rb:1748:in `block in visit_heredoc'
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/prism-0.19.0/lib/prism/translation/parser/compiler.rb:1746:in `each'
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/prism-0.19.0/lib/prism/translation/parser/compiler.rb:1746:in `visit_heredoc'
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/prism-0.19.0/lib/prism/translation/parser/compiler.rb:938:in `visit_interpolated_string_node'
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/prism-0.19.0/lib/prism/node.rb:10263:in `accept'
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/prism-0.19.0/lib/prism/compiler.rb:29:in `visit'
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/prism-0.19.0/lib/prism/translation/parser/compiler.rb:283:in `visit_call_node'
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/prism-0.19.0/lib/prism/node.rb:2308:in `accept'
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/prism-0.19.0/lib/prism/compiler.rb:34:in `block in visit_all'
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/prism-0.19.0/lib/prism/compiler.rb:34:in `map'
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/prism-0.19.0/lib/prism/compiler.rb:34:in `visit_all'
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/prism-0.19.0/lib/prism/translation/parser/compiler.rb:1463:in `visit_statements_node'
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/prism-0.19.0/lib/prism/node.rb:16077:in `accept'
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/prism-0.19.0/lib/prism/translation/parser/compiler.rb:1287:in `visit_program_node'
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/prism-0.19.0/lib/prism/node.rb:14158:in `accept'
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/prism-0.19.0/lib/prism/translation/parser.rb:132:in `build_ast'
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/prism-0.19.0/lib/prism/translation/parser.rb:47:in `parse'
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/parser-3.3.0.5/lib/parser/base.rb:33:in `parse'
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/parser-3.3.0.5/lib/parser/base.rb:65:in `parse_file'
        from -e:1:in `<main>'
```

**Comment by koic** [comment]
I made a mistake regarding the expected behavior. It should be raised `Parser::SyntaxError`, not `ArgumentError`:

## Expected

```console
$ ruby -rparser/current -e 'Parser::CurrentRuby.parse_file("heredoc_with_force_encoding.rb")'
(string):3:1-4:0: error: literal contains escape sequences incompatible with UTF-8
(string):3:   # Euro symbol: \xa4
(string):3: ^~~~~~~~~~~~~~~~~~~~~
(string):4: RUBY
(string):4:
/Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/parser-3.3.0.5/lib/parser/diagnostic/engine.rb:72:in `process': literal contains escape sequences incompatible with UTF-8 (Parser::SyntaxError)
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/parser-3.3.0.5/lib/parser/builders/default.rb:2303:in `diagnostic'
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/parser-3.3.0.5/lib/parser/builders/default.rb:2291:in `string_value'
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/parser-3.3.0.5/lib/parser/builders/default.rb:325:in `string_internal'
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/parser-3.3.0.5/lib/parser/ruby33.rb:11744:in `_reduce_595'
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/racc-1.7.3/lib/racc/parser.rb:267:in `_racc_do_parse_c'
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/racc-1.7.3/lib/racc/parser.rb:267:in `do_parse'
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/parser-3.3.0.5/lib/parser/base.rb:190:in `parse'
        from /Users/koic/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/parser-3.3.0.5/lib/parser/base.rb:33:in `parse'
        from -e:1:in `<main>'
```

The difference in these exceptions causes an error in RuboCop. The reason is that `ArgumentError` occurs contrary to the `Parser::SyntaxError` that RuboCop AST expects:
https://github.com/rubocop/rubocop-ast/blob/v1.30.0/lib/rubocop/ast/processed_source.rb#L217

This exception compatibility will likely be necessary for integration with RuboCop.

**Comment by kddnewton** [comment]
@koic I think this should be fixed in v0.21.0, can you test it out?

**Comment by koic** [comment]
Great! This issue has been resolved in v0.21.0. Thank you!
